potodo/_github.py

```diff
@@ -1,32 +1,56 @@
+import re
 import requests
 
-from subprocess import getoutput
+from subprocess import check_output
 
 
-def get_repo_url():
-    url = getoutput("git remote get-url --all upstream")
+def get_repo_url() -> str:
+    """
+    Tries to get the repository url from git commands
+    """
+    url: str = str(check_output("git remote get-url --all upstream"))
     if "fatal" in url:
-        url = getoutput("git remote get-url --all origin")
+        url = str(check_output("git remote get-url --all origin"))
     if "fatal" in url:
+        # If the commands didn't work
         raise ValueError(
             f"Unknown error. `git get-url --all upstream|origin` returned {url}"
         )
     return url
 
 
-def get_repo_name():
+def get_repo_name() -> str:
+    """
+    Will get the repository url from git commands then remove useless stuff to get ORG/NAME
+    """
     repo_url: str = get_repo_url()
-    repo_url = repo_url.replace("https://github.com/", "")
-    repo_url = repo_url.replace("git@github.com:", "")
-    repo_name = repo_url.replace(".git", "")
+    # Removes useless stuff. If it isn't there then nothing happens
+    repo_url = repo_url.strip("https://github.com/")
+    repo_url = repo_url.strip("git@github.com:")
+    repo_name = repo_url.strip(".git")
     return repo_name
 
 
 def get_reservation_list():
-    issues = requests.get(
-        "https://api.github.com/repos/" + get_repo_name() + "/issues"
-    ).json()
-    reservations = {
-        issue["title"].split()[-1].lower(): issue["user"]["login"] for issue in issues
-    }
+    resp = requests.get(
+        "https://api.github.com/repos/" + get_repo_name() + "/issues?state=open"
+    )
+    issues = resp.json()
+
+    if resp.headers.get('Link'):
+        link = resp.headers.get('Link')
+        last_page = link.split(">")[0]
+        for page in range(2, int(last_page) + 1):
+            resp = requests.get(
+                "https://api.github.com/repos/" + "python/python-docs-fr" + "/issues?state=open&page={}".format(page)
+            )
+            issues + resp.json()
```

Comment:
It would be RESTfuller to simply follow the given next, like this:
```python
next = "https://api.github.com/repos/" + "python/python-docs-fr" + "/issues?state=open"
while next:
    resp = requests.get(next)
    issues += resp.json()
    next = get_next_from_link(resp.headers['Link'])
```

To parse the link you should use something like:

```pycon
>>> requests.get('https://api.github.com/repos/python/python-docs-fr/issues?state=open').links["next"]["url"]
'https://api.github.com/repositories/92097535/issues?state=open&page=2'
```
Or even:
```pycon
requests.get('https://api.github.com/repos/python/python-docs-fr/issues?state=open').links.get("next", {}).get("url")
```
so you get the expected `None` at the last page.

Reply:
Done